COMPONENT-GUIDELINES.md

```diff
@@ -0,0 +1,293 @@
+# Purpose
+
+This document defines guidelines and patterns developers should follow when building components for the Pipedream registry.
+
+Developers may create, deploy and share components that do not conform to these guidelines, but they will not be eligible to be listed in the curated registry (e.g., they may be hosted in a Github repo). If you develop a component that does not adhere to these guidelines, but you believe there is value to the broader community, please [reach out in our community forum](https://pipedream.com/community). We can either provide guidance to improve compliance or evaluate if an exception is warranted.
```

Comment:
In the first sentence, maybe link "components" to the components section below so if new users are reading this, they're clear on what a component is?

```
[components](#components)
```

Reply:
When you link to the community, maybe link directly to https://pipedream.com/community/c/dev/11 ?